tests/sentry/workflow_engine/endpoints/test_organization_available_action_index.py

```diff
@@ -0,0 +1,407 @@
+from dataclasses import dataclass
+from unittest.mock import ANY, patch
+
+from sentry.constants import SentryAppStatus
+from sentry.plugins.base.manager import PluginManager
+from sentry.plugins.sentry_webhooks.plugin import WebHooksPlugin
+from sentry.testutils.cases import APITestCase
+from sentry.testutils.silo import region_silo_test
+from sentry.utils.registry import Registry
+from sentry.workflow_engine.handlers.action.notification.base import IntegrationActionHandler
+from sentry.workflow_engine.models.action import Action
+from sentry.workflow_engine.types import ActionHandler
+from sentry_plugins.pagerduty.plugin import PagerDutyPlugin
+from sentry_plugins.slack.plugin import SlackPlugin
+from sentry_plugins.trello.plugin import TrelloPlugin
+
+
+@region_silo_test
+class OrganizationAvailableActionAPITestCase(APITestCase):
+    endpoint = "sentry-api-0-organization-available-action-index"
+
+    def setUp(self):
+        super().setUp()
+        self.login_as(user=self.user)
+
+        self.registry = Registry[ActionHandler](enable_reverse_lookup=False)
+        self.registry_patcher = patch(
+            "sentry.workflow_engine.endpoints.organization_available_action_index.action_handler_registry",
+            new=self.registry,
+        )
+        self.registry_patcher.start()
+
+        self.plugin_registry = PluginManager()
+        self.plugins_registry_patcher = patch(
+            "sentry.workflow_engine.processors.action.plugins",
+            new=self.plugin_registry,
+        )
+        self.plugins_registry_patcher.start()
+
+    def tearDown(self) -> None:
+        super().tearDown()
+        self.registry_patcher.stop()
+        self.plugins_registry_patcher.stop()
+
+    def test_simple(self):
+        @self.registry.register(Action.Type.EMAIL)
+        @dataclass(frozen=True)
+        class EmailActionHandler(ActionHandler):
+            group = ActionHandler.Group.NOTIFICATION
+            config_schema = {}
+            data_schema = {}
+
+        response = self.get_success_response(
+            self.organization.slug,
+            status_code=200,
+        )
+        assert len(response.data) == 1
+        assert response.data == [
+            {
+                "type": Action.Type.EMAIL,
+                "handlerGroup": ActionHandler.Group.NOTIFICATION.value,
+                "configSchema": {},
+                "dataSchema": {},
+            }
+        ]
+
+    def test_integrations(self):
+        @self.registry.register(Action.Type.MSTEAMS)
+        @dataclass(frozen=True)
+        class MSTeamsActionHandler(IntegrationActionHandler):
```

Comment:
What's the purpose of including this if we only make a Slack integration?

Reply:
This tests that integration actions with no installations do not get returned. I'll add a comment :)